dvc/command/experiments.py

```diff
@@ -41,18 +41,18 @@ def _filter(filters, update_func):
                     raise InvalidArgumentError(
                         f"'{name}' does not match any known {label}"
                     )
-                update_func(matches)
+                update_func({match: None for match in matches})
 
     if include:
-        ret = set()
+        ret = OrderedDict()
```

Comment:
Use `OrderedDict()` with `None` values to get set-like behavior with preserved ordering

Reply:
Or, use a `dict`?

Reply:
I thought the preserved key ordering for regular dict technically isn't guaranteed before 3.7, so I thought it would be preferred to just use `OrderedDict`?

Reply:
yes, you are right that it is only on `>3.6`. My point was that, as CPython and Python are closely tied, no one is going to change that behavior in 3.6, so should be safe to just use `dict` for convenience. Saying that I have no issues with OrderedDict, it was just a suggestion.